diff -uraN snacc-1.3bbn.orig/snacc-config.in snacc-1.3bbn/snacc-config.in
--- snacc-1.3bbn.orig/snacc-config.in	2001-01-27 02:02:40.000000000 +0100
+++ snacc-1.3bbn/snacc-config.in	2008-04-16 01:19:43.153755670 +0200
@@ -95,7 +95,7 @@
     if test $exec_prefix != /usr; then echo -n "-L${exec_prefix}/lib "; fi
     case "$language" in
     C)   echo -n "-lasn1c${buffertype} " ;;
-    C++) echo -n "-Iasn1c++ " ;;
+    C++) echo -n "-lasn1c++ " ;;
     *) ;;
     esac
   else
